.github/workflows/checks-vizro-ai.yml

```diff
@@ -36,10 +36,10 @@ jobs:
           python-version: ${{ env.PYTHON_VERSION }}
 
       - name: Install Hatch
-        run: pip install --upgrade hatch
+        run: pip install hatch
 
-      - name: List dependencies
-        run: hatch run all.py${{ env.PYTHON_VERSION }}:pip freeze
+      - name: Show dependency tree
+        run: hatch run pip tree
```

Comment:
Maybe a stupid question but: "What's `pip tree`?"

Reply:
It's the `uv` equivalent of `pipdeptree`. Basically it gives not just a flat list of dependencies but also the chain showing where they come from. This is very useful in debugging CI failures when one of your transitive dependencies has changed and broken something and you're trying to work out why you're using it. Like this:
```
selenium v4.2.0
├── urllib3 v1.26.20
├── trio v0.26.2
│   ├── attrs v24.2.0
│   ├── sortedcontainers v2.4.0
│   ├── idna v3.10
│   ├── outcome v1.3.0.post0
│   │   └── attrs v24.2.0
│   └── sniffio v1.3.1
└── trio-websocket v0.11.1
    ├── trio v0.26.2 (*)
    └── wsproto v1.2.0
        └── h11 v0.14.0
```